crates/bevy_ecs/src/component.rs

```diff
@@ -171,7 +171,7 @@ pub struct ComponentDescriptor {
     // SAFETY: This must remain private. It must only be set to "true" if this component is
     // actually Send + Sync
     is_send_and_sync: bool,
-    type_id: Option<TypeId>,
+    type_id: TypeId,
```

Comment:
This was intentionally an Option because "dynamic" / "scripted" components will not have a TypeId

Reply:
Ah..., makes sense then, I will revert the last commit